#! /bin/sh /usr/share/dpatch/dpatch-run
## 15_munge_8bit.dpatch by Sven Mueller <debian@incase.de>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Add support for munge8bit

@DPATCH@
--- a/imap/message.c
+++ b/imap/message.c
@@ -227,6 +227,7 @@ unsigned size;
     int n;
     int sawcr = 0, sawnl;
     int reject8bit = config_getswitch(IMAPOPT_REJECT8BIT);
+    int munge8bit = config_getswitch(IMAPOPT_MUNGE8BIT);
     int inheader = 1, blankline = 1;
 
     while (size) {
@@ -262,7 +263,7 @@ unsigned size;
 			/* We have been configured to reject all mail of this
 			   form. */
 			if (!r) r = IMAP_MESSAGE_CONTAINS8BIT;
-		    } else {
+		    } else if (munge8bit) {
 			/* We have been configured to munge all mail of this
 			   form. */
 			*p = 'X';
--- a/imap/spool.c
+++ b/imap/spool.c
@@ -140,6 +140,7 @@ static int parseheader(struct protstream
     state s = NAME_START;
     int r = 0;
     int reject8bit = config_getswitch(IMAPOPT_REJECT8BIT);
+    int munge8bit = config_getswitch(IMAPOPT_MUNGE8BIT);
     const char **skip = NULL;
 
     if (namelen == 0) {
@@ -266,7 +267,7 @@ static int parseheader(struct protstream
 			   form. */
 			r = IMAP_MESSAGE_CONTAINS8BIT;
 			goto ph_error;
-		    } else {
+		    } else if (munge8bit) {
 			/* We have been configured to munge all mail of this
 			   form. */
 			c = 'X';
--- a/lib/imapoptions
+++ b/lib/imapoptions
@@ -469,6 +469,11 @@ are listed with ``<none>''.
 { "mboxlist_db", "skiplist", STRINGLIST("flat", "berkeley", "berkeley-hash", "skiplist")}
 /* The cyrusdb backend to use for the mailbox list. */
 
+{ "munge8bit", 1, SWITCH }
+/* If enabled, lmtpd changes 8-bit characters to `X'. Also see reject8bit.
+   (A proper soultion to non-ASCII characters in headers is offered by  
+   RFC 2047 and its predecessors.) */
+
 # xxx badly worded
 { "mupdate_connections_max", 128, INT }
 /* The max number of connections that a mupdate process will allow, this
@@ -703,9 +708,9 @@ are listed with ``<none>''.
 
 { "reject8bit", 0, SWITCH }
 /* If enabled, lmtpd rejects messages with 8-bit characters in the
-   headers.  Otherwise, 8-bit characters are changed to `X'.  (A
-   proper soultion to non-ASCII characters in headers is offered by  
-   RFC 2047 and its predecessors.) */
+   headers. Also see munge8bit, which is only applied if reject8bit is
+   not activated. (A proper soultion to non-ASCII characters in headers
+   is offered by RFC 2047 and its predecessors.) */
 
 { "rfc2046_strict", 0, SWITCH }
 /* If enabled, imapd will be strict (per RFC 2046) when matching MIME
